File: zhaoyali/singleNumber.c
//
//  singleNumber.c
//  LianJiaShell
//
//  Created by bailee on 2017/7/31.
//  Copyright © 2017年 bailee. All rights reserved.
//

#include <stdio.h>
int singleNumber(int* nums, int numsSize) {
    int res = 0;
    for (int i = 0; i< numsSize; i++) {
        res = res ^nums[i];
    }
    return res;
}

File: caoguodong/Single Number.java
public class Solution {
    public int singleNumber(int[] nums) {
        if(nums.length<=1){
            return nums[0];
        }
        
        Arrays.sort(nums);
        
        for(int i=1; i<nums.length; i=i+2){
            if(nums[i] != nums[i-1]){
                return nums[i-1];
            }
            
            if(i+2>=nums.length){
                return nums[i+1];
            }
        }
        
        return 0;
    }
}